### PR TITLE
Create Tree_Postorder_Traversal.kt

### LINKED ISSUE #2551: Tree Postorder Traversal
- [x] C
- [x] C++
- [ ] C# - @sharmakeshav10
- [ ] CoffeeScript
- [ ] Dart - @Amitsharma45
- [x] Go
- [x] Java
- [x] JS
- [ ] Kotlin - @PrachieNaik
- [x] Php
- [x] Python
- [ ] Readme - @SimranMakhija7
- [x] Ruby
- [ ] Rust
- [ ] TypeScript - @shivamgupta57121

### PR DESCRIPTION
**Pull request by PrachieNaik**
Fixes #2551

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Languages Used:
Kotlin

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import java.util.*
```
Comment (asha15) [inline Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt:1]: keep a line space after this

File: Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+}
+
+fun main() {
+    var read = Scanner(System.`in`)
+    println("Enter the size of Array:")
+    val arrSize = read.nextLine().toInt()
+    var arr = IntArray(arrSize)
+    val nodes = mutableListOf<Node<Int>>()
+    println("Enter the array respresentaion of binary tree")
```
Comment (asha15) [inline Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt:22]: keep a line space after this

File: Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    for(i in 0 until arrSize)
+    {
+        arr[i] = read.nextLine().toInt()
+        nodes.add(Node(arr[i]))
+    }
```
Comment (asha15) [inline Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt:27]: keep a line space after this

File: Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    for(i in 0..arrSize-2)
+    {
+        if((i*2)+1<arrSize && arr[(i*2)+1]!=-1)
+        nodes[i].left = nodes[(i*2)+1]
+        if((i*2)+2<arrSize && arr[(i*2)+2]!=-1)
+        nodes[i].right = nodes[(i*2)+2]
```
Comment (asha15) [inline Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt:33]: keep space around operators

File: Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    }
+    print("Post Order traversal of tree is ")
+    nodes[0].postorderTraversal()
+}
```
Comment (asha15) [inline Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt:37]: keep a line space after this

**Comment by PrachieNaik** [comment]
@asha15 Can you please review now?

File: Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    val nodes = mutableListOf<Node<Int>>()
+    println("Enter the array respresentaion of binary tree")
+    
+    for(i in 0 until arrSize)
+    {
+        arr[i] = read.nextLine().toInt()
+        nodes.add(Node(arr[i]))
+    }
+    
+    for(i in 0..arrSize-2)
```
Comment (asha15) [inline Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt:31]: please keep space around operator

Comment (PrachieNaik) [reply Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt]: Can you please check now? I will squash it from next time

**Comment by PrachieNaik** [comment]
Can someone review and merge it? @asha15 @jainaman224

**Comment by PrachieNaik** [comment]
@asha15 Please check now

File: Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	    nodes[i].left = nodes[(i * 2) + 1]
+	}
+	    
+	if ((i * 2) + 2 < arrSize && arr[(i * 2) + 2] != -1)
+	{    
+	    nodes[i].right = nodes[(i * 2) + 2]
+	}
+    }
+    print("Post Order traversal of tree is ")
+    nodes[0].postorderTraversal()
```
Comment (asha15) [inline Tree_Postorder_Traversal/Tree_Postorder_Traversal.kt:44]: correct the indentation

**Comment by asha15** [comment]
please squash your commits before push it

**Comment by PrachieNaik** [comment]
@asha15 Please review

**Comment by somya-kapoor** [comment]
please squash your commits

**Comment by PrachieNaik** [comment]
can you guys please use squash and merge option? Thank you

**Pull request merged**